pkg/sidecar/pod.go

```diff
@@ -71,6 +82,16 @@ func remove(pod corev1.Pod) corev1.Pod {
 		}
 	}
 	pod.Spec.Containers = containers
+
+	// NOTE: we also remove init containers (native sidecars) since k8s 1.28.
+	// This should have no side effects.
+	var initContainers []corev1.Container
+	for _, initContainer := range pod.Spec.InitContainers {
+		if initContainer.Name != naming.Container() {
+			initContainers = append(initContainers, initContainer)
+		}
+	}
+	pod.Spec.InitContainers = initContainers
```

Comment:
How about using https://pkg.go.dev/slices#DeleteFunc? Also, shouldn't this only happen if the feature flag is set?